cns/kubecontroller/nodenetworkconfig/conversion.go

```diff
@@ -87,13 +91,41 @@ func CreateNCRequestFromStaticNC(nc v1alpha.NetworkContainer) (*cns.CreateNetwor
 
 	secondaryIPConfigs := map[string]cns.SecondaryIPConfig{}
 
-	// iterate through all IP addresses in the subnet described by primaryPrefix and
-	// add them to the request as secondary IPConfigs.
-	for addr := primaryPrefix.Masked().Addr(); primaryPrefix.Contains(addr); addr = addr.Next() {
-		secondaryIPConfigs[addr.String()] = cns.SecondaryIPConfig{
-			IPAddress: addr.String(),
-			NCVersion: int(nc.Version),
+	switch nc.Type {
+	case v1alpha.VNETBlock:
+		{
+			for blockIndex, ipAssignment := range nc.IPAssignments {
+				cidrPrefix, err := netip.ParsePrefix(ipAssignment.IP)
+				if err != nil {
+					return nil, errors.Wrapf(ErrInvalidCidrIPBlock, "CIDR IP Block: %s", ipAssignment.IP)
+				}
+
+				zeroAddr := cidrPrefix.Masked().Addr()
+				if blockIndex == 0 {
+					zeroAddr = zeroAddr.Next() // # Block 1 CIDR. Exclude 1st IP from pod assignment
+				}
+
+				// iterate through all IP addresses in the CIDR block described by cidrPrefix and
+				// add them to the request as secondary IPConfigs.
+				for addr := zeroAddr; cidrPrefix.Contains(addr); addr = addr.Next() {
+					secondaryIPConfigs[addr.String()] = cns.SecondaryIPConfig{
+						IPAddress: addr.String(),
+						NCVersion: int(nc.Version),
+					}
+				}
+			}
+		}
+	case v1alpha.Overlay:
+		// iterate through all IP addresses in the subnet described by primaryPrefix and
+		// add them to the request as secondary IPConfigs.
+		for addr := primaryPrefix.Masked().Addr(); primaryPrefix.Contains(addr); addr = addr.Next() {
+			secondaryIPConfigs[addr.String()] = cns.SecondaryIPConfig{
+				IPAddress: addr.String(),
+				NCVersion: int(nc.Version),
+			}
 		}
+	case v1alpha.VNET:
+		return nil, errors.New("vnet mode not supported for static NC")
```

Comment:
we don't need to check this, this function is only called when we have already checked that the AssignmentMode is `Static`

Reply:
VNETScale involves adding/removing SecondaryIPs to the NNC based on demand, though. Isn't that actually `Dynamic`?

Reply:
Vnet Scale will be static allocation mode

Reply:
My understanding was we assign all CIDR blocks in NNC during initialization and then there are no updates.
@nairashu Can u confirm?

Reply:
Updated to remove unnecessary case.